astroquery/simbad/core.py

```diff
@@ -936,6 +1016,256 @@ def query_objectids_async(self, object_name, *, cache=True,
+        looks for columns in the selected Simbad tables that contain the
+        given keyword. The keyword search is not case-sensitive.
+
+        Parameters
+        ----------
+        *tables : str, optional
+            Add tables names as strings to restrict to these tables columns.
+        keyword : str, optional
+            A keyword to look for in column names, table names, or descriptions.
+        get_adql : bool, optional
+            Returns the ADQL string instead of querying SIMBAD.
+
+        Examples
+        --------
+        >>> from astroquery.simbad import Simbad
+        >>> Simbad.list_columns("ids", "ident") # doctest: +REMOTE_DATA
+        <Table length=4>
+        table_name column_name datatype ...  unit    ucd
+          object      object    object  ... object  object
+        ---------- ----------- -------- ... ------ -------
+             ident          id  VARCHAR ...        meta.id
+             ident      oidref   BIGINT ...
+               ids         ids     CLOB ...        meta.id
+               ids      oidref   BIGINT ...
+
+        >>> from astroquery.simbad import Simbad
+        >>> Simbad.list_columns(keyword="filter") # doctest: +REMOTE_DATA
+        <Table length=5>
+         table_name column_name   datatype  ...  unit           ucd
+           object      object      object   ... object         object
+        ----------- ----------- ----------- ... ------ ----------------------
+             filter description UNICODECHAR ...        meta.note;instr.filter
+             filter  filtername     VARCHAR ...                  instr.filter
+             filter        unit     VARCHAR ...                     meta.unit
+               flux      filter     VARCHAR ...                  instr.filter
+        mesDiameter      filter        CHAR ...                  instr.filter
+
+        >>> from astroquery.simbad import Simbad
+        >>> Simbad.list_columns("basic", keyword="object") # doctest: +REMOTE_DATA
+        <Table length=4>
+        table_name column_name datatype ...  unit          ucd
+          object      object    object  ... object        object
+        ---------- ----------- -------- ... ------ -------------------
+             basic     main_id  VARCHAR ...          meta.id;meta.main
+             basic   otype_txt  VARCHAR ...                  src.class
+             basic         oid   BIGINT ...        meta.record;meta.id
+             basic       otype  VARCHAR ...                  src.class
+        """
+        query = ("SELECT table_name, column_name, datatype, description, unit, ucd"
+                 " FROM TAP_SCHEMA.columns"
+                 " WHERE table_name NOT LIKE 'TAP_SCHEMA.%'")
+        # select the tables
+        if len(tables) == 1:
+            query += f" AND table_name = '{tables[0]}'"
+        elif len(tables) > 1:
+            query += f" AND table_name IN ({str(tables)[1:-1]})"
```

Comment:
I don't really get this logic without some more background info, how many `tables` are expected here? Why drop the first and last ones? Could you maybe add a one-line comment or reference above the line?

Reply:
Hum it's useless. I was removing the parenthesis around the string representation of the tuple and then I was adding them around again.

Reply:
And for the other question, tables is a tuple with zero or more items.

```python
from astroquery.simbad import Simbad
Simbad.list_columns()
```

will return all columns of all tables while 

```python
Simbad.list_columns("ref")
```

will returns columns of the `ref` table only and 

```python
Simbad.list_columns("cat", "ident", "ids")
```

will return columns of `cat`, `ids` and `ident`